### PR TITLE
Update patch release docs for v1.14

### PR DESCRIPTION
**Pull request by spiffxp**
See individual commits for details.

Part 1 is the last of https://github.com/kubernetes/sig-release/issues/428

Part 2 is updating patch release team info for v1.14

### CODE REVIEW

**Comment by spiffxp** [comment]
ref: https://github.com/kubernetes/sig-release/issues/369

**Comment by spiffxp** [comment]
/area release-team
since this has to do with patch release management

**Comment by tpepper** [comment]
This is going to conflict with #534

File: releases/patch-releases.md [hunk 1]
```diff
@@ -74,7 +112,8 @@ days after cherry-picks deadline to allow all release-blocking tests to run.
 
 | **Patch Release** | **Cherry-picks deadline** | **Target date** | **Owner** |
 | --- | --- | --- | --- |
-| 1.12.6 | TBA | TBA | TBA |
+| 1.12.7 | TBA | TBA | @feiskyer |
```
Comment (feiskyer) [inline releases/patch-releases.md]: also update the date for next releases? I think we've agreed to cut a set of patch releases next monday

**Comment by spiffxp** [comment]
d'oh and I even reviewed that... yyyes this absolutely will conflict, do you want to amend that PR to include 1.14, or shall I wait until it merges and de-conflict?

**Comment by tpepper** [comment]
> d'oh and I even reviewed that... yyyes this absolutely will conflict, do you want to amend that PR to include 1.14, or shall I wait until it merges and de-conflict?

It's ready for de-conflicting...will +1 here when you have.

**Pull request merged**